accepted/future-releases/nnbd/feature-specification.md

```diff
@@ -217,7 +220,16 @@ assignable to `Object`.
 
 It is not an error for the body of a `late` field to reference `this`.
 
-It is an error for a formal parameter to be declared `late`.
+It is an error for a variable to be declared as `late` in any of the following
+positions: in a formal parameter list of any kind; in a catch clause; in the
+variable binding section of a c-style `for` loop, a `for in` loop, an `await
+for` loop, or a `for element` in a collection literal.
+
+It is an error for the initializer expression of a `late` local variable to use
+a prefix `await` expression.
```

Comment:
It is an error in this case:
```dart
main() async {
   late v = await 42;
}
```

But is it an error in this case?
```dart
main() async {
  late v = () async {
    await 42;
  };
}
```

Reply:
No.

So,  maybe phrase it as:

> It is a compile-time error if the initializer expression of a `late` local variable contains an `await` expression which is not inside another function expressions.